### PR TITLE
adjusted c#, xml, axml and xaml

### PR DESCRIPTION
**Pull request by eberthold**
Adjusted colors for the Frappé theme based on the official style guide. Including more C# colors, VsVim colors, XML, AXML and XAML.

I'm not exactly shure about the xml coloring, because it currently off between c# doc comments and plain xml. Is a xml object a class or a keyword?

Here is an example Screenshot
image

relates to #12

### CODE REVIEW

**Comment by djflan** [comment]
@eberthold I don't think the xml doc is treated like standard xml-> it is [compiler generated](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/).. it is more or less, proto-xml. 

There might in fact be a way to change this specific markup, you can try my "tool" [here](https://github.com/djflan/Visual-Studio-Theme-Tools) if you want to find the correct name for the element. This is the process I use to find the aberrant ones for Catppuccin today. 

I have not yet taken a look at your additions, but should be able to this coming weekend. Thank you for your contribution!

**Comment by djflan** [comment]
@eberthold Any chance for a before/after shot (CS)? For some reason, I can't seem to get the changes to take effect on my system.

Also, I should mention, we normally need PRs to incorporate changes to the other themes as well. I realize this is still a work in progress, but I did want to mention that.

**Comment by eberthold** [comment]
Sure, here you go. Left is before, right is after.
It now should be following the [styleguide](https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md#code-editors) more closely

image

I'm currently writing a small powershell script to port the changes from on theme to another. If it's done I will update this PR and set it to ready for review

**Comment by eberthold** [comment]
The crucial parts which needed to be adjusted were the following
image

Those elements can be found in the editor by using the "All elements" tab and grouping by color instead of category to know if one is changing the foreground or background of said element
image


There are may be some changes or nuances missing as stuff like interfaces and properties are not mentioned in the styleguide.
If I'm totally off in any regards, feel free to abandon this PR

**Comment by djflan** [comment]
@eberthold Looks great

**Comment by eberthold** [comment]
So I added a script to convert from one theme to another. So only one theme needs to be edited manually. I converted Latte, Macchiato and Mocca with this script and on a quick look it seemed right.

Please have a look for yourself if it matches.

**Comment by djflan** [comment]
@eberthold I couldn't get any of the themes in your branch to build. My simple guess here is that the order of the 'Background' node matters (i.e. It should appear before the foreground). 

Builds:
`Build started...
1>------ Build started: Project: Catppuccin for Visual Studio, Configuration: Debug Any CPU ------
Restored C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\Catppuccin for Visual Studio.csproj (in 4 ms).
1>  Microsoft(R) Visual Studio(R) Vsix Color Compiler version 17.0.0.1.
1>  (C) Microsoft Corporation. All rights reserved.
1>  Microsoft(R) Visual Studio(R) Vsix Color Compiler version 17.0.0.1.
1>  (C) Microsoft Corporation. All rights reserved.
1>  Error occurred when reading xml file 'C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\Catppuccin Latte.vstheme': Error occured parsing XML: The element 'Color' has invalid child element 'Background'.
1>  Line: 10896
1>C:\Users\djfla\.nuget\packages\microsoft.visualstudio.vsixcolorcompiler\17.0.31709.430\build\Microsoft.VisualStudio.VsixColorCompiler.targets(47,5): error MSB3073: The command ""C:\Users\djfla\.nuget\packages\microsoft.visualstudio.vsixcolorcompiler\17.0.31709.430\build\\..\VsixColorCompiler.exe" "C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\Catppuccin Latte.vstheme" "bin\Debug\Catppuccin Latte.pkgdef" /registerTheme" exited with code 13.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build started at 8:54 AM and took 01.374 seconds ==========

Build started...
1>------ Build started: Project: Catppuccin for Visual Studio, Configuration: Debug Any CPU ------
Restored C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\Catppuccin for Visual Studio.csproj (in 3 ms).
1>  Microsoft(R) Visual Studio(R) Vsix Color Compiler version 17.0.0.1.
1>  (C) Microsoft Corporation. All rights reserved.
1>  Microsoft(R) Visual Studio(R) Vsix Color Compiler version 17.0.0.1.
1>  (C) Microsoft Corporation. All rights reserved.
1>  Error occurred when reading xml file 'C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\Catppuccin Macchiato.vstheme': Error occured parsing XML: The element 'Color' has invalid child element 'Background'.
1>  Line: 11049
1>C:\Users\djfla\.nuget\packages\microsoft.visualstudio.vsixcolorcompiler\17.0.31709.430\build\Microsoft.VisualStudio.VsixColorCompiler.targets(47,5): error MSB3073: The command ""C:\Users\djfla\.nuget\packages\microsoft.visualstudio.vsixcolorcompiler\17.0.31709.430\build\\..\VsixColorCompiler.exe" "C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\Catppuccin Macchiato.vstheme" "bin\Debug\Catppuccin Macchiato.pkgdef" /registerTheme" exited with code 13.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build started at 9:00 AM and took 01.331 seconds ==========

Build started...
1>------ Build started: Project: Catppuccin for Visual Studio, Configuration: Debug Any CPU ------
1>  Microsoft(R) Visual Studio(R) Vsix Color Compiler version 17.0.0.1.
1>  (C) Microsoft Corporation. All rights reserved.
1>  Microsoft(R) Visual Studio(R) Vsix Color Compiler version 17.0.0.1.
1>  (C) Microsoft Corporation. All rights reserved.
1>  Error occurred when reading xml file 'C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\Catppuccin Mocha.vstheme': Error occured parsing XML: The element 'Color' has invalid child element 'Background'.
1>  Line: 7732
1>C:\Users\djfla\.nuget\packages\microsoft.visualstudio.vsixcolorcompiler\17.0.31709.430\build\Microsoft.VisualStudio.VsixColorCompiler.targets(47,5): error MSB3073: The command ""C:\Users\djfla\.nuget\packages\microsoft.visualstudio.vsixcolorcompiler\17.0.31709.430\build\\..\VsixColorCompiler.exe" "C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\Catppuccin Mocha.vstheme" "bin\Debug\Catppuccin Mocha.pkgdef" /registerTheme" exited with code 13.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build started at 9:27 AM and took 01.291 seconds ==========

Build started...
1>------ Build started: Project: Catppuccin for Visual Studio, Configuration: Debug Any CPU ------
1>  Microsoft(R) Visual Studio(R) Vsix Color Compiler version 17.0.0.1.
1>  (C) Microsoft Corporation. All rights reserved.
1>  Microsoft(R) Visual Studio(R) Vsix Color Compiler version 17.0.0.1.
1>  (C) Microsoft Corporation. All rights reserved.
1>  Error occurred when reading xml file 'C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\_Catppuccin Latte.vstheme': Error occured parsing XML: The element 'Color' has invalid child element 'Background'.
1>  Line: 10896
1>C:\Users\djfla\.nuget\packages\microsoft.visualstudio.vsixcolorcompiler\17.0.31709.430\build\Microsoft.VisualStudio.VsixColorCompiler.targets(47,5): error MSB3073: The command ""C:\Users\djfla\.nuget\packages\microsoft.visualstudio.vsixcolorcompiler\17.0.31709.430\build\\..\VsixColorCompiler.exe" "C:\temp\cr\catppuccin-visual-studio\Catppuccin VS Themes\_Catppuccin Latte.vstheme" "bin\Debug\_Catppuccin Latte.pkgdef" /registerTheme" exited with code 13.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build started at 9:28 AM and took 01.298 seconds ==========`

File: Catppuccin VS Themes/Catppuccin Frappé.vstheme [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <Themes>
-  <Theme Name="Catppuccin Frappé" GUID="{217075d3-6c1c-432d-9587-ca060f7229fe}" BaseGUID="{1ded0138-47ce-435e-84ef-9ec1f439b749}">
+  <Theme Name="Catppuccin Frappé (7)" GUID="{f215a980-7da4-45b3-8020-0785d4e29ee2}" BaseGUID="{1ded0138-47ce-435e-84ef-9ec1f439b749}">
```
Comment (djflan) [inline Catppuccin VS Themes/Catppuccin Frappé.vstheme:2]: Theme GUID and names should not change.

File: Catppuccin VS Themes/Catppuccin Latte.vstheme [hunk 2]
```diff
@@ -10848,5 +10860,1921 @@
...
+      <Color Name="parameter name">
+        <Foreground Type="CT_RAW" Source="FFE64553" />
+        <Background Type="CT_INVALID" Source="00000000" />
+      </Color>
+      <Color Name="delegate name">
+        <Foreground Type="CT_RAW" Source="FFFE640B" />
+        <Background Type="CT_INVALID" Source="00000000" />
+      </Color>
+      <Color Name="enum name">
+        <Foreground Type="CT_RAW" Source="FFFE640B" />
```
Comment (djflan) [inline Catppuccin VS Themes/Catppuccin Latte.vstheme:10930]: I believe foreground nodes prior to background nodes might be causing build issues. Please see my comment about this.

File: Catppuccin VS Themes/ThemeConvert.ps1
Comment (djflan) [inline Catppuccin VS Themes/ThemeConvert.ps1:1]: Nice addition here. Lets put it under repo root in '.dev' folder

**Review by djflan — CHANGES_REQUESTED** [review]
Need to solve the build issues. 
Theme guids and names need to be preserved.

File: Catppuccin VS Themes/Catppuccin Frappé.vstheme (hunk 1 above)
Comment (eberthold) [reply Catppuccin VS Themes/Catppuccin Frappé.vstheme:2]: fixed - that's a weird behavior of the vs theming extension when you try to review manual changes by applying those directly from the designer editor...

**Review by djflan — APPROVED** [review]

**Comment by djflan** [comment]
@eberthold Thank you for your contribution. I think there might still be some small things to clean up ('try'/'catch' and 'return' seem to be red instead of purple), but I think this can be done at a later date.

Thank you for all your hard work.

**Comment by eberthold** [comment]
> 'try'/'catch' and 'return' seem to be red instead of purple

@djflan 
That was a decision I made purposely because most themes use a different color for normal keywords and control flow keywords to better grasp the flow of the code.

The color name is "keyword - control" if you'd like to change it. Maybe there is a more fitting color in the palette to distinguish the flow keyword from the normal C# keywords and the parameters as well (which are now very close and nearly indistinguishable)

**Comment by djflan** [comment]
@eberthold Currently looking over things. So far, things look good.
I like the idea of control flow statements being a different color. I think the red is fine

**Pull request merged**